smarts/core/coordinates.py

```diff
@@ -46,13 +48,28 @@ def as_lwh(self):
 class Heading(float):
     """In this space we use radians, 0 is facing north, and turn counter-clockwise."""
 
+    def __init__(self, value=...):
+        float.__init__(value)
+
+    def __new__(self, x: Union[SupportsFloat, SupportsIndex] = ...) -> "Heading":
+        """A override to constrain heading to -pi to pi"""
+        value = x
+        if isinstance(value, (int, float)):
+            value = value % (2 * math.pi)
+            if value > math.pi:
+                value -= 2 * math.pi
+            if value < -math.pi:
+                value += 2 * math.pi
```

Comment:
I think `%` only returns non-negative value, so perhaps the check for `value < -math.pi` is not needed.

Reply:
You are right, I intended `math.fmod()` here.